### PR TITLE
[functorch] Remove not needed named member polyfill functions

### PR DESCRIPTION
**Pull request by XuehaiPan**
The `nn.Module` APIs already support `remove_duplicate` argument. It's time to retire these not needed polyfill functions. They are identical to the `nn.Module.named_parameters` and `nn.Module.named_buffers` methods.

### CODE REVIEW

File: torch/_functorch/named_members_polyfill.py [hunk 1]
```diff
@@ -1,32 +0,0 @@
-# Polyfilled from pytorch core while we figure out the `remove_duplicate` issues.
```
Comment (XuehaiPan) [inline torch/_functorch/named_members_polyfill.py:1]: This is supported in `nn.Module` now.

**Review by ezyang — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**